yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1278,6 +1374,74 @@ def pixelize_sph_kernel_arbitrary_grid(np.float64_t[:, :, :] buff,
                         continue
                     buff[xi, yi, zi] += buff_num[xi, yi, zi] / buff_denom[xi, yi, zi]
 
+@cython.initializedcheck(False)
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def pixelize_sph_kernel_gather_arbitrary_grid(np.float64_t[:, :, :] buff,
+        np.int64_t[:, :, :, :] pids, np.float64_t[:, :, :, :] dists,
+        np.float64_t[:] hsml, np.float64_t[:] pmass, np.float64_t[:] pdens,
+        np.float64_t[:] quantity_to_smooth, PyKDTree tree=None, np.int64_t
+        offset=0, kernel_name="cubic", use_normalization=True):
+
+    cdef np.intp_t xsize, ysize, zsize
+    cdef np.float64_t w_j, coeff
+    cdef np.int64_t xi, yi, zi, pi
+    cdef np.float64_t q, h_j2, ih_j2
+    cdef int count, i, j, k, particle
+    cdef np.float64_t[:, :, :] buff_num
+    cdef np.float64_t[:, :, :] buff_denom
+    cdef np.int64_t[:] tree_id
+
+    if tree is not None:
+        tree_id = tree.idx.astype("int64")
+
+    xsize, ysize, zsize = buff.shape[0], buff.shape[1], buff.shape[2]
+    if use_normalization:
+        buff_num = np.zeros((xsize, ysize, zsize), dtype='f8')
+        buff_denom = np.zeros((xsize, ysize, zsize), dtype='f8')
+
+    kernel_func = get_kernel_func(kernel_name)
+
+    # define this to avoid using the use_normalization python object in the
+    # tight loop
+    cdef np.intp_t use_norm = int(use_normalization)
+
+    with nogil:
+        for xi in range(xsize):
+            for yi in range(ysize):
+                for zi in prange(zsize):
```

Comment:
regarding the use of `prange` here - are you not worried about https://github.com/cython/cython/issues/2316 anymore?

Reply:
yeah, thank you